framework/base/Security.php

```diff
@@ -90,7 +90,39 @@ class Security extends Component
      * @since 2.0.6
      */
     public $passwordHashCost = 13;
+    /**
+     * @var boolean The recent (> 2.1.5) LibreSSL RNGs are faster and likely better than /dev/urandom.
+     */
+    private $_useLibreSSL;
+
+
+    /**
+     * Check that openssl used LibreSSL  version >= 2.1.5
+     * @return boolean that used LibreSSL version 2.1.5 or hi
+     */
+    public function getUseLibreSSL()
+    {
+        if ($this->_useLibreSSL === null) {
+            // Parse OPENSSL_VERSION_TEXT because OPENSSL_VERSION_NUMBER is no use for LibreSSL.
+            // https://bugs.php.net/bug.php?id=71143
+            $this->_useLibreSSL = defined('OPENSSL_VERSION_TEXT')
+                && preg_match('{^LibreSSL (\d\d?)\.(\d\d?)\.(\d\d?)$}', OPENSSL_VERSION_TEXT, $matches)
+                && (10000 * $matches[1]) + (100 * $matches[2]) + $matches[3] >= 20105;
+        }
 
+        return $this->_useLibreSSL;
+    }
+
+    /**
+     * Check server operation system - windows or not
+     * @return boolean true if used windows on server
+     */
+    public static function isWindows()
```

Comment:
I don't think it is a good idea to make it part of the public API.

Reply:
ExposedSecurity.php is a good idea for this case? Make method protected in Security and public in ExposedSecurity.php

Reply:
Yes.

Reply:
No, only this. Which call do you prefer

Reply:
?